src/main/java/seedu/address/model/recipe/Recipe.java

```diff
@@ -83,9 +83,11 @@ public ServingSize getServingSize() {
      * @return true if both recipes have the same name.
      */
     public boolean isSameRecipe(Recipe otherRecipe) {
-        return otherRecipe == this
-                || (otherRecipe != null
-                && otherRecipe.getName().equals(this.getName()));
+        if (otherRecipe == null) {
+            return false;
+        }
+
+        return otherRecipe.getName().equals(this.getName());
```

Comment:
Yeah ^

Reply:
Wah this one a bit weird. When I was implementing it earlier yesterday IntelliJ was throwing some error and was refusing to build, but now it works AHHAHA, I'll push the change now that it can build properly 👍